tests/app/notifications/test_rest.py

```diff
@@ -182,6 +182,49 @@ def test_get_all_notifications_for_job_in_order_of_job_number(notify_api,
             assert response.status_code == 200
 
 
+def test_get_all_notifications_for_job_filtered_by_status(
+    notify_api,
+    notify_db,
+    notify_db_session,
+    sample_service
+):
+    with notify_api.test_request_context():
+        with notify_api.test_client() as client:
+            job = create_sample_job(notify_db, notify_db_session, service=sample_service)
+
+            create_sample_notification(
+                notify_db,
+                notify_db_session,
+                job=job,
+                to_field="1",
+                created_at=datetime.utcnow(),
+                status=NOTIFICATION_STATUS_TYPES[0],
+                job_row_number=1
+            )
+
+            for length_expected, status in [
+                (1, NOTIFICATION_STATUS_TYPES[0]),
+                (0, NOTIFICATION_STATUS_TYPES[1])
+            ]:
```

Comment:
i spent far too long last night on the train home working out how you could do it with enumerate

(for reference `for length_expected, status in enumerate(reversed(NOTIFICATION_STATUS_TYPES[1::-1])):` *should* work)